packages/cypress/src/executors/cypress/cypress.impl.ts

```diff
@@ -154,30 +154,37 @@ async function* startDevServer(
     return;
   }
 
-  const { project, target, configuration } = parseTargetString(
-    opts.devServerTarget
-  );
-  const devServerTargetOpts = readTargetOptions(
-    { project, target, configuration },
-    context
-  );
-  const targetSupportsWatchOpt =
-    Object.keys(devServerTargetOpts).includes('watch');
-
-  for await (const output of await runExecutor<{
-    success: boolean;
-    baseUrl?: string;
-  }>(
-    { project, target, configuration },
-    // @NOTE: Do not forward watch option if not supported by the target dev server,
-    // this is relevant for running Cypress against dev server target that does not support this option,
-    // for instance @nguniversal/builders:ssr-dev-server.
-    targetSupportsWatchOpt ? { watch: opts.watch } : {},
-    context
-  )) {
-    if (!output.success && !opts.watch)
-      throw new Error('Could not compile application files');
-    yield opts.baseUrl || (output.baseUrl as string);
+  const devServerTargets = opts.devServerTarget.split(/\s*,\s*/);
```

Comment:
when moving to string array this shouldn't be needed. you can normalize the option to always be a string array so you don't have to check if it's a string or string[]

Reply:
I can make it a string array if needed. Will also add unit and e2e tests later.